test/spec/esSearchSvc.js

```diff
@@ -411,6 +411,117 @@ describe('Service: searchSvc: ElasticSearch', function() {
     });
   });
 
+  describe('vars', function() {
+    it('replaces vars no URI encode', function() {
+      var mockQueryText = 'taco&burrito';
+      var mockEsParams  = {
+        query: {
+          term: {
+            text: '#$query##'
+          }
+        }
+      };
+      var searcher = searchSvc.createSearcher(
+        mockFieldSpec.fieldList,
+        mockEsUrl,
+        mockEsParams,
+        mockQueryText,
+        {},
+        'es'
+      );
+      $httpBackend.expectPOST(mockEsUrl, function verifyDataSent(data) {
+        var esQuery = angular.fromJson(data);
+        return (esQuery.query.term.text === mockQueryText);
+      }).
+      respond(200, mockResults);
+      searcher.search();
+      $httpBackend.flush();
+      $httpBackend.verifyNoOutstandingExpectation();
+    });
+
+    it('replaces keywords vars', function() {
+      var mockQueryText = 'taco&burrito purina headphone';
+      var mockEsParams  = {
+        query: {
+          term: {
+            text: '#$query1## #$query## #$query2##'
+          }
+        }
+      };
+      var searcher = searchSvc.createSearcher(
+        mockFieldSpec.fieldList,
+        mockEsUrl,
+        mockEsParams,
+        mockQueryText,
+        {},
+        'es'
+      );
+      $httpBackend.expectPOST(mockEsUrl, function verifyDataSent(data) {
+        var esQuery = angular.fromJson(data);
+        return (esQuery.query.term.text === 'taco&burrito taco&burrito purina headphone purina');
```

Comment:
Is it better to call the variable #$keyword1##

?

-Doug

On Wed, Dec 2, 2015 at 9:45 AM, Youssef Chaker <notifications@github.com>
wrote:

> In test/spec/esSearchSvc.js
> <https://github.com/o19s/splainer-search/pull/22#discussion_r46421685>:
>
> > +          term: {
> > +            text: '#$query1## #$query## #$query2##'
> > +          }
> > +        }
> > +      };
> > +      var searcher = searchSvc.createSearcher(
> > +        mockFieldSpec.fieldList,
> > +        mockEsUrl,
> > +        mockEsParams,
> > +        mockQueryText,
> > +        {},
> > +        'es'
> > +      );
> > +      $httpBackend.expectPOST(mockEsUrl, function verifyDataSent(data) {
> > +        var esQuery = angular.fromJson(data);
> > +        return (esQuery.query.term.text === 'taco&burrito taco&burrito purina headphone purina');
>
> tricky!
> spent 10 minutes trying to figure out how we got this form, only to
> realize that the middle one is #$query## which is replaced by the full
> query string!!!
> got cross [image: :eyes:] in the process
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/o19s/splainer-search/pull/22/files#r46421685>.
>